crates/sui-framework/packages/bridge/sources/limiter.move

```diff
@@ -0,0 +1,192 @@
+        while (i > 0 && record.last_recorded_hour < (current_hour_since_epoch - 24 + i)) {
+            record.total_amount = record.total_amount - vector::remove(&mut record.per_hour_amounts, 0);
+            i = i - 1;
+        };
+        // Backfill missing hours
+        while (record.last_recorded_hour < current_hour_since_epoch) {
+            // only backfill up to 24 hours
+            if (record.last_recorded_hour < current_hour_since_epoch - 23) {
+                record.last_recorded_hour = current_hour_since_epoch - 23;
+            }else {
+                record.last_recorded_hour = record.last_recorded_hour + 1;
+            };
+            vector::push_back(&mut record.per_hour_amounts, 0);
+        };
+
+        // Get limit for the route
+        let route_limit = if (vec_map::contains(&self.transfer_limits, &route)) {
+            *vec_map::get(&self.transfer_limits, &route)
+        } else {
+            DEFAULT_TRANSFER_LIMIT
+        };
+
+        // Compute notional amount
+        let coin_type = type_name::get<T>();
+        let notional_amount = *vec_map::get(&self.notional_values, &coin_type) * amount;
+        let notional_amount = notional_amount / pow(10, treasury::token_decimals<T>());
+
+        // Check if transfer amount exceed limit
+        assert!(record.total_amount + notional_amount <= route_limit, ETransferAmountExceedLimit);
+
+        // Record transfer value
+        let new_amount = vector::pop_back(&mut record.per_hour_amounts) + notional_amount;
+        vector::push_back(&mut record.per_hour_amounts, new_amount);
+        record.total_amount = record.total_amount + notional_amount;
+    }
+
+    #[test]
+    fun test_24_hours_windows() {
+        let limiter = TransferLimiter {
+            transfer_limits: vec_map::empty(),
+            notional_values: vec_map::empty(),
+            // Per hour transfer amount for each bridge route
+            transfer_records: vec_map::empty(),
+        };
+
+        let route = chain_ids::get_route(chain_ids::sui_devnet(), chain_ids::eth_sepolia());
+
+        // Global transfer limit is 1M USD
+        vec_map::insert(&mut limiter.transfer_limits, route, 1_000_000);
+        // Notional price for ETH is 10 USD
+        vec_map::insert(&mut limiter.notional_values, type_name::get<ETH>(), 10);
+
+        let scenario = test_scenario::begin(@0x1);
+        let ctx = test_scenario::ctx(&mut scenario);
+        let clock = clock::create_for_testing(ctx);
+        clock::set_for_testing(&mut clock, 1706288001377);
+
+        check_and_record_transfer<ETH>(&clock, &mut limiter, route, 1_000_000_000_000);
+
+        let record = vec_map::get(&limiter.transfer_records, &route);
+        assert!(record.total_amount == 10000 * 10, 0);
+
+        // transfer 1000 ETH every hour, the 24 hours totol should be 24000 * 10
+        let i = 0;
+        while (i < 50) {
+            clock::increment_for_testing(&mut clock, 60 * 60 * 1000);
```

Comment:
can we use different values such as `1, 2, 3, 4...` instead of `10`s, which is hopefully harder to miss issues